controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -621,106 +632,130 @@ def _prepare_mocks_for_get_host_by_identifiers_(self, result_reader_iter, hosts)
+    def test_get_host_by_identifiers_no_other_ports_return_iscsi_host(self, result_reader_iter):
+        host_with_iqn_list = self._get_host_as_dictionary('host_id_s', 'test_host_s', iscsi_names_list=['iqn.test.s'])
+        self._prepare_mocks_for_get_host_by_identifiers(result_reader_iter, special_host=host_with_iqn_list)
+        hostname, connectivity_types = self.svc.get_host_by_host_identifiers(
+            Initiators('Test_nqn', ['Test_wwn'], 'iqn.test.s'))
+        self.assertEqual('test_host_s', hostname)
+        self.assertEqual([config.ISCSI_CONNECTIVITY_TYPE], connectivity_types)
 
     @patch("controller.array_action.svc_cli_result_reader.SVCListResultsReader.__iter__")
     def test_get_host_by_identifiers_return_iscsi_host_with_list_iqn(self, result_reader_iter):
-        host_1 = self._get_host_as_dictionary('host_id_1', 'test_host_1', [], ['abc1'])
-        host_2 = self._get_host_as_dictionary('host_id_2', 'test_host_2', ['iqn.test.2', 'iqn.test.22'], ['abc3'])
-        host_3 = self._get_host_as_dictionary('host_id_3', 'test_host_3', ['iqn.test.3'], ['abc3'])
-        hosts = [host_1, host_2, host_3]
-        self._prepare_mocks_for_get_host_by_identifiers_(result_reader_iter, hosts)
-        host, connectivity_type = self.svc.get_host_by_host_identifiers(Initiators(
-            'iqn.test.2', ['abcd3']))
-        self.assertEqual('test_host_2', host)
-        self.assertEqual([config.ISCSI_CONNECTIVITY_TYPE], connectivity_type)
+        host_with_iqn_list = self._get_host_as_dictionary('host_id_s', 'test_host_s', wwpns_list=['wwns'],
+                                                          iscsi_names_list=['iqn.test.s1', 'iqn.test.s2'])
+        self._prepare_mocks_for_get_host_by_identifiers(result_reader_iter, special_host=host_with_iqn_list)
+        hostname, connectivity_types = self.svc.get_host_by_host_identifiers(
+            Initiators('Test_nqn', ['Test_wwn'], 'iqn.test.s1'))
+        self.assertEqual('test_host_s', hostname)
+        self.assertEqual([config.ISCSI_CONNECTIVITY_TYPE], connectivity_types)
+
+    @patch("controller.array_action.svc_cli_result_reader.SVCListResultsReader.__iter__")
+    def test_get_host_by_identifiers_return_nvme_host(self, result_reader_iter):
+        self._prepare_mocks_for_get_host_by_identifiers(result_reader_iter)
+        hostname, connectivity_types = self.svc.get_host_by_host_identifiers(
+            Initiators('nqn.test.3', ['Test_wwn'], 'iqn.test.6'))
+        self.assertEqual('test_host_3', hostname)
+        self.assertEqual([config.NVME_OVER_FC_CONNECTIVITY_TYPE], connectivity_types)
+
+    @patch("controller.array_action.svc_cli_result_reader.SVCListResultsReader.__iter__")
+    def test_get_host_by_identifiers_no_other_ports_return_nvme_host(self, result_reader_iter):
+        host_with_iqn_list = self._get_host_as_dictionary('host_id_s', 'test_host_s', nqn_list=['nqn.test.s'])
```

Comment:
```suggestion
        host_with_nqn_list = self._get_host_as_dictionary('host_id_s', 'test_host_s', nqn_list=['nqn.test.s'])
```

Reply:
yup, done

Reply:
I still see this gap in other places (e.g. the sloppy link).
please don't rely only on my examples - I might miss some.
the best strategy is for you to first go over all places, and only then I will go over all places as well.